### PR TITLE
random: Base-like APIs for rand, rand!, randn, randn!

### PR DESCRIPTION
**Pull request by iblis17**
and deprecate the original APIs

e.g.
```julia
julia> mx.rand(2, 3)
2×3 mx.NDArray{Float32,2} @ CPU0:
 0.631961  0.324175  0.0762663
 0.285366  0.395292  0.074995

julia> mx.rand(2, 3, low = 1, high = 10)
2×3 mx.NDArray{Float32,2} @ CPU0:
 7.83884  7.85793  7.64791
 7.68646  8.56082  8.42189
```

```julia
julia> mx.randn(2, 3)
2×3 mx.NDArray{Float32,2} @ CPU0:
 0.962853  0.424535  -0.320123
 0.478113  1.72886    1.72287

julia> mx.randn(2, 3, μ = 100)
2×3 mx.NDArray{Float32,2} @ CPU0:
 99.5635  100.483   99.888
 99.9889  100.533  100.072
```

### CODE REVIEW

**Comment by iblis17** [comment]
maybe this is not a good idea,
I cannot figure out how to specifying the rand in this case:
```julia
julia> Base.rand(-0.1:0.1, 2, 3)
2×3 Array{Float64,2}:
 -0.1  -0.1  -0.1 
 -0.1  -0.1  -0.1
```

**Comment by iblis17** [comment]
ha, it's `Base.rand()*0.2 - 0.1`

**Comment by iblis17** [comment]
changed `mx.rand(1:10, 2, 3)` -> `mx(2, 3, low = 1, high = 10)`

File: src/deprecated.jl [hunk 1]
```diff
@@ -23,3 +23,29 @@
...
+function randn(μ, σ, dims::NTuple{N,Int}, ctx::Context = cpu()) where N
+  warn("mx.randn(μ, σ, dims, ctx = cpu()) is deprecated, use " *
+       "mx.randn(dims...; μ = μ, σ = σ, context = ctx) instead.")
+  mx.randn(dims...; μ = μ, σ = σ, context = ctx)
+end
+
+function randn!(μ, σ, x::NDArray)
+  warn("mx.randn!(μ, σ, x::NDArray) is deprecated, use " *
+       "mx.randn!(x; μ = 0, σ = 1) instead.")
+  randn!(x; μ = 0, σ = 1)
```
Comment (pluskid) [inline src/deprecated.jl]: do you mean `μ = μ, σ = σ`?

Comment (iblis17) [reply src/deprecated.jl]: oh, yeah

**Pull request merged**